### ISSUE TITLE
Dependency problem on gatsby-transformer-sharp + npm install 

### ISSUE THREAD

**Issue by pawsong** [gatsbyjs/gatsby#4507]
## Description

`npm run build` failed.

### Steps to reproduce

On [this repository](https://github.com/AustinGreen/gatsby-starter-netlify-cms)

```bash
$ npm install
$ npm run build
```

Throws

```bash
Error: Cannot find module 'sharp'
```

Error

### Expected result

It should not throw error.

### Actual result

It throws error.

### Environment

* Gatsby version (`npm list gatsby`): 1.9.215
* gatsby-cli version (`gatsby --version`): 1.1.45
* Node.js version: v8.9.3
* Operating System: MacOS

If packages are installed with `yarn istall`, everything goes well as https://github.com/gatsbyjs/gatsby/blob/master/yarn.lock#L12329 installs 'sharp' module under `<root>/node_modules`. But `npm install` installs `sharp` under `<root>/node_modules/gatsby-plugin-sharp`, which cannot be accessed from `<root>/node_modules/gatsby-transformer-sharp`, so it breaks the build.

I think the root cause of this issue is that `gatsby-transformer-sharp`'s [package.json](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-sharp/package.json) does not include `sharp` even when [it uses `sharp` module](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-sharp/src/extend-node-type.js#L19).

**Comment by m-allanson** [comment]
Thanks @pawsong, I can replicate this issue. 

Some of Gatsby's plugins have moved to using peerDependencies (e.g. [`gatsby-plugin-styled-components`](https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-styled-components#breaking-changes-history)). @KyleAMathews should we do that for `gatsby-transformer-sharp` and `gatsby-plugin-sharp`?

**Comment by KyleAMathews** [comment]
Yeah that'd be great. I think @jlengstorf did a bunch of research on this recently.

**Comment by 68696c6c** [comment]
I had this same error.  It seems to have been fixed by running `npm install sharp`

**Comment by giraffesyo** [comment]
For me what fixed the error was installing gatsby-plugin-sharp. Even though, I'm not using it directly with `gatsby-transformer-sharp` it seems to have been required. We should update the documentation to mention this.

**Comment by jlengstorf** [comment]
This was fixed in #6487
